src/NuGet.Clients/NuGet.PackageManagement.UI/Models/PackageSolutionDetailControlModel.cs

```diff
@@ -201,24 +201,28 @@ private async Task UpdateInstalledVersionsAsync(CancellationToken cancellationTo
 
         protected override async Task CreateVersionsAsync(CancellationToken cancellationToken)
         {
-            _versions = new List<DisplayVersion>();
-            var allVersions = _allPackageVersions?.Where(v => v.version != null).OrderByDescending(v => v);
-
             // allVersions is null if server doesn't return any versions.
-            if (allVersions == null || !allVersions.Any())
+            if (_allPackageVersions == null || _allPackageVersions.Count == 0)
             {
                 return;
             }
 
+            _versions = new List<DisplayVersion>();
+            var allVersions = _allPackageVersions?.Where(v => v.version != null).OrderByDescending(v => v.version.Version).ToList();
+
             // null, if no version constraint defined in package.config
             VersionRange allowedVersions = await GetAllowedVersionsAsync(cancellationToken);
             var allVersionsAllowed = allVersions.Where(v => allowedVersions.Satisfies(v.version)).ToArray();
 
             // null, if all versions are allowed to install or update
-            var blockedVersions = allVersions
-                .Select(v => v.version)
-                .Where(v => !allVersionsAllowed.Any(allowed => allowed.version.Equals(v)))
-                .ToArray();
+            var blockedVersions = new List<NuGetVersion>(allVersions.Count);
+            foreach (var version in allVersions)
+            {
+                if (!allVersionsAllowed.Any(a => a.version.Version.Equals(version.version.Version)))
```

Comment:
`version.version.Version` looks absurd.  Is the type of `version` [here](https://github.com/NuGet/NuGet.Client/pull/3833/files#diff-5c15ee840d9063ec5f345761223159c7ecde0f058e0e0190e359adcfee761290R219) actually `Version`?  Maybe the variable can be more appropriately named.